tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/default/field_default_persist_test.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/code_analysis_collector.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var config = GeneratorConfigBuilder().build();
+  group('Given a class with fields with a "defaultPersist" keyword', () {
+    test(
+      'when the field is of type DateTime and the default is set to "now", then the field should have a "default persist" value',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+            class: Example
+            table: example
+            fields:
+              dateTimeType: DateTime?, defaultPersist=now
+            ''',
+          ).build()
+        ];
+
+        var collector = CodeGenerationCollector();
+        var definitions =
+            StatefulAnalyzer(config, models, onErrorsCollector(collector))
+                .validateAll();
+
+        expect(collector.errors, isEmpty);
+
+        var definition = definitions.first as ClassDefinition;
+        expect(definition.fields.last.defaultPersistValue, 'now');
+      },
+    );
+
+    test(
+      'when the field is of type DateTime and the default is set to UTC format string, then the field should have a "default persist" value',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+          class: Example
+          table: example
+          fields:
+            dateTimeType: DateTime, defaultPersist=2024-05-34T22:00:00.000Z
```

Comment:
Question: is the Z at the end required for this string to be valid? According to the test description it seem so, would be nice with a test case for the string `2024-05-34T22:00:00.000`